static/app/components/group/externalIssuesList/streamlinedExternalIssueList.tsx

```diff
@@ -112,143 +104,152 @@ export function StreamlinedExternalIssueList({
-
-            return (
-              <ErrorBoundary key={integration.key} mini>
-                <DropdownMenu
-                  trigger={triggerProps => (
-                    <IssueActionDropdownMenu
-                      {...sharedButtonProps}
-                      {...triggerProps}
-                      showChevron={false}
-                    />
-                  )}
-                  items={integration.actions.map(action => ({
-                    key: action.id,
-                    ...getActionLabelAndTextValue({
-                      action,
-                      integrationDisplayName: integration.displayName,
-                    }),
-                    onAction: action.onClick,
-                    disabled: integration.disabled,
-                  }))}
-                />
-              </ErrorBoundary>
-            );
-          })}
-        </IssueActionWrapper>
-      )}
+          }}
+          options={props.integrations.map(integration => ({
+            key: integration.key,
+            disabled: integration.disabled,
+            leadingItems: (
+              <Flex align="center" justify="center" style={{minHeight: 19}}>
+                {integration.displayIcon}
+              </Flex>
+            ),
+            tooltip: integration.disabled ? integration.disabledText : undefined,
+            label: integration.displayName,
+            hideCheck: true,
+            value: integration.key,
+            textValue: integration.key,
+            details:
+              integration.actions.length > 1 ? (
+                <ExternalIssueSubmenu integration={integration} />
+              ) : undefined,
+            showDetailsInOverlay: true,
+          }))}
+        />
+      </CompositeSelect>
     </Fragment>
   );
 }
 
-const IssueActionWrapper = styled('div')`
-  display: flex;
-  flex-wrap: wrap;
-  gap: ${space(1)};
-  line-height: 1.2;
-`;
+function ExternalIssueSubmenu(props: {integration: ExternalIssueIntegration}) {
+  const {integration} = props;
+  const {overlayState} = useContext(SelectContext);
+  return integration.actions.map(action => {
+    const itemProps: MenuListItemProps = {
+      tooltip: action.disabled ? action.disabledText : undefined,
+      disabled: action.disabled,
+      ...getActionLabelAndTextValue({
+        action,
+        integrationDisplayName: integration.displayName,
+      }),
+    };
+    const callbackProps: Record<string, () => void> = {
+      onPointerDown: () => {
+        overlayState?.close();
+        action.onClick();
+      },
+    };
+    return <MenuListItem key={action.id} {...callbackProps} {...itemProps} />;
+  });
+}
 
-const LinkedIssue = styled(LinkButton)`
-  display: flex;
-  align-items: center;
-  padding: ${space(0.5)} ${space(0.75)};
-  border: 1px solid ${p => p.theme.border};
-  border-radius: ${p => p.theme.borderRadius};
-  font-weight: normal;
-`;
+function ExternalIssueMenuEmpty() {
+  return (
+    <Flex
+      style={{padding: space(3)}}
```

Comment:
not sure I like the inline style here. I guess this where we usually did styled components? What about `<Flex padding=“lg”>` ? We could add more base props as we find out that we need them, or add them all (padding, margin, backgroundColor, border, ...) which brings us back to a Box model 😅

Reply:
Lets chat about that, I was doing some exploration and I think we have a good way forward.